js_modules/dagster-ui/packages/ui-core/src/automation/MergedAutomationRoot.tsx

```diff
@@ -110,13 +109,38 @@ export const MergedAutomationRoot = () => {
     );
   }, [cachedData, visibleRepos]);
 
+  const tagsFilter = useDefinitionTagFilterWithManagedState({
+    allTags: useTagsForObjects(
+      repoBuckets,
+      useCallback((repoBucket) => {
+        return [
+          ...repoBucket.schedules.flatMap((schedule) => schedule.tags),
+          ...repoBucket.sensors.flatMap((sensor) => sensor.tags),
+        ];
+      }, []),
+    ),
+  });
+  const {state: tagFilterState} = tagsFilter;
+
+  const filters = useMemo(
+    () => [codeLocationFilter, runningStateFilter, automationTypeFilter, tagsFilter],
+    [codeLocationFilter, runningStateFilter, automationTypeFilter, tagsFilter],
+  );
+  const {button: filterButton, activeFiltersJsx} = useFilters({filters});
+
   const {state: runningState} = runningStateFilter;
 
   const filteredBuckets = useMemo(() => {
     return repoBuckets.map(({sensors, schedules, ...rest}) => {
       return {
         ...rest,
-        sensors: sensors.filter(({sensorState}) => {
+        sensors: sensors.filter(({sensorState, tags}) => {
+          if (
+            tagFilterState.size &&
+            !doesFilterArrayMatchValueArray(Array.from(tagFilterState), tags)
```

Comment:
Just a drive-by nit but it's not clear what "match" means here without clicking into the definition. Maybe "doesValueContainAnyFilter" or "doesValueEqualFilter"?